packages/web-app-files/src/views/spaces/Project.vue

```diff
@@ -288,12 +288,12 @@ export default {
   },
   watch: {
     $route: {
-      handler: function (to, from) {
-        const sameRoute = to.name === from?.name
+      handler: async function (to, from) {
+        const sameRoute = to.name === from?.name && to.params?.storageId === from?.params?.storageId
```

Comment:
I took the liberty of making `from.params` optional - `from` is not guaranteed to be defined.